src/solvers/hard_coded_ars343.jl

```diff
@@ -117,44 +104,35 @@ function step_u!(integrator, cache::IMEXARKCache, ::ARS343)
         dt * a_exp[i, 3] * T_exp[3] +
         dt * a_imp[i, 2] * T_imp[2] +
         dt * a_imp[i, 3] * T_imp[3]
-    post_explicit!(U, p, t_exp)
-
+    dss!(U, p, t_exp)
     @. temp = U # used in closures
     let i = i
         t_imp = t + dt * c_imp[i]
+        post_implicit!(U, p, t_imp)
         implicit_equation_residual! = (residual, Ui) -> begin
             T_imp!(residual, Ui, p, t_imp)
             @. residual = temp + dt * a_imp[i, i] * residual - Ui
         end
         implicit_equation_jacobian! = (jacobian, Ui) -> begin
             T_imp!.Wfact(jacobian, Ui, p, dt * a_imp[i, i], t_imp)
         end
-        call_post_implicit! = Ui -> begin
-            post_implicit!(Ui, p, t_imp)
-        end
-        call_post_implicit_last! = Ui -> begin
-            dss!(Ui, p, t_imp)
-            post_implicit!(Ui, p, t_imp)
-        end
+        call_post_implicit! = Ui -> post_implicit!(Ui, p, t_imp)
         solve_newton!(
             newtons_method,
             newtons_method_cache,
             U,
             implicit_equation_residual!,
             implicit_equation_jacobian!,
             call_post_implicit!,
-            call_post_implicit_last!,
+            nothing,
         )
+        @. T_imp[i] = (U - temp) / (dt * a_imp[i, i])
+        dss!(U, p, t_imp)
+        post_explicit!(U, p, t_imp)
     end
-
-    @. T_imp[i] = (U - temp) / (dt * a_imp[i, i])
-
     T_lim!(T_lim[i], U, p, t_exp)
     T_exp!(T_exp[i], U, p, t_exp)
 
-    # final
-    i = -1
```

Comment:
I think I intentionally did this so that any future access would be an OOB error